File: C#/C# 1/3. Operators-Expressions-Statements-Homework/Point-In-A-Circle/PointInACircle.cs
/* Problem 7. Point in a Circle
   Write a program that reads the coordinates of a point x and y and using an expression checks if given point (x, y) is inside a circle K({0, 0}, 2) - the center has coordinates (0, 0) and the circle has radius 2. The program should then print "yes DISTANCE" if the point is inside the circle or "no DISTANCE" if the point is outside the circle. In place of DISTANCE print the distance from the beginning of the coordinate system - (0, 0) - to the given point. The distance should be always printed with 2-digit precision after the floating point. */

using System;
using System.Threading;                                                     
using System.Globalization;

class PointInACircle
{
    static void Main()
    {
        Thread.CurrentThread.CurrentCulture = new CultureInfo("en-US");     
        
        double x = double.Parse(Console.ReadLine());        
        double y = double.Parse(Console.ReadLine());

        double distance = Math.Sqrt(x * x + y * y); // using Pythagorean theorem

        Console.WriteLine((distance <= 2) ? "yes {0:F2}" : "no {0:F2}", distance);               
    }
}

File: Front-end/JS Fundamentals/operators and expressions/06. Point in Circle/point-in-a-circle.js
function solve(args) {
    var x = +args[0];
    var y = +args[1];

    var distance = Math.sqrt(x * x + y * y);

    // first way
    if (distance <= 2) {
        console.log('yes ' + distance.toFixed(2));
    } else {
        console.log('no ' + distance.toFixed(2));
    }

    // other way
    // var output = distance <= 2 ? 'yes ' + distance.toFixed(2) : 'no ' + distance.toFixed(2);
    // console.log(output);
}

solve([-2, 0]);
solve([-1, 2]);
solve([1.5, -1]);
solve([-1.5, -1.5]);
solve([100, -30]);
solve([0, 0]);
solve([0.2, -0.8]);
solve([0.9, -1.93]);
solve([1, 1.655]);
solve([0, 1]);